vmm/src/vm.rs

```diff
@@ -953,41 +952,68 @@ impl Vm {
     }
 
     #[cfg(target_arch = "aarch64")]
-    fn load_kernel(&mut self) -> Result<EntryPoint> {
-        let guest_memory = self.memory_manager.lock().as_ref().unwrap().guest_memory();
+    fn load_firmware(mut firmware: &File, device_manager: Arc<Mutex<DeviceManager>>) -> Result<()> {
+        // uefi_flash may not be ready. If not, wait for it.
+        while device_manager
+            .lock()
+            .as_ref()
+            .unwrap()
+            .uefi_flash()
+            .is_none()
+        {
+            std::thread::sleep(std::time::Duration::from_millis(1));
```

Comment:
This is completely the wrong way to solve such a data race.

Reply:
Thanks @rbradford -, I think that eliminating the dependency of device_manager by moving uefi_flash to memory_manager as @michael2012z says can avoid this. WDYT?

Reply:
Okay, I think that will work.